python/samples/concepts/agents/step4_openai_assistant_agent_code_interpreter.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft. All rights reserved.
+import asyncio
+import os
+
+from semantic_kernel.agents.open_ai.azure_open_ai_assistant_agent import AzureOpenAIAssistantAgent
+from semantic_kernel.agents.open_ai.open_ai_assistant_agent import OpenAIAssistantAgent
+from semantic_kernel.agents.open_ai.open_ai_assistant_definition import OpenAIAssistantDefinition
+from semantic_kernel.agents.open_ai.open_ai_service_configuration import (
+    AzureOpenAIServiceConfiguration,
+    OpenAIServiceConfiguration,
+)
+from semantic_kernel.agents.open_ai.open_ai_thread_creation_options import OpenAIThreadCreationOptions
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.kernel import Kernel
+
+AGENT_NAME = "CodeRunner"
+AGENT_INSTRUCTIONS = "Run the provided code file and return the result."
+
+# Note: you may toggle this to switch between AzureOpenAI and OpenAI
+use_azure_openai = True
+
+
+# A helper method to invoke the agent with the user input
+async def invoke_agent(agent: OpenAIAssistantAgent, thread_id: str, input: str) -> None:
+    """Invoke the agent with the user input."""
+    await agent.add_chat_message(thread_id=thread_id, message=ChatMessageContent(role=AuthorRole.USER, content=input))
+
+    print(f"# {AuthorRole.USER}: '{input}'")
+
+    async for content in agent.invoke(thread_id=thread_id):
+        if content.role != AuthorRole.TOOL:
+            print(f"# {content.role}: {content.content}")
+
+
+async def main():
+    # Create the instance of the Kernel
+    kernel = Kernel()
+
+    # Add the OpenAIChatCompletion AI Service to the Kernel
+    service_id = "agent"
+
+    # Create the agent
+    definition = OpenAIAssistantDefinition(
+        name=AGENT_NAME, instructions=AGENT_INSTRUCTIONS, enable_code_interpreter=True
+    )
+
+    # Create the agent
+    if use_azure_openai:
+        configuration = AzureOpenAIServiceConfiguration.create(service_id=service_id)
+        agent = AzureOpenAIAssistantAgent(kernel=kernel, configuration=configuration, definition=definition)
+    else:
+        configuration = OpenAIServiceConfiguration.create(service_id=service_id)
+        agent = OpenAIAssistantAgent(kernel=kernel, configuration=configuration, definition=definition)
+
+    await agent.create_assistant()
+
+    code_file_path = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+        "resources",
+        "agent_assistant_code_interpreter_file",
+        "fibonacci.py",
```

Comment:
The code intepreter should be generating the fibonacci code.  No need to upload.  Ping me if you have question on this sample.

Reply:
I wanted to show an example of how one would manage a code file that the interpreter runs. I can update to have the assistant generate and run the code, no problem.

Reply:
I updated this to be similar to the dotnet sample.